### PR TITLE
Add Trino 391 release notes

### LINKED ISSUE #13187: Release notes for 391

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 391 release.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Docs only.

> How would you describe this change to a non-technical end user or system administrator?

Add the release notes for Trino 391 with sections for general, security, connectors and other topics as included in the release.

## Related issues, pull requests, and links

Fixes #13187 

## Documentation

( ) No documentation is needed.
(✅) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(✅) No release notes entries required.
( ) Release notes entries required with the following suggested text:

## Verification for each commit

Format: commit, committer/maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 15 July 2022 ✅ 

5fcfaf30a64583633d3531b8e4305ddbe058a796 @ebyhr #13182 ✅ rn ✅ docs
0e8d068e073da65e34937086c5249c0cbfeef528 @ebyhr #13188 ✅ rn ✅ docs
0104c20d435148e0b69b08216b91daf6b428a5f0 @ebyhr #13190 ✅ rn ✅ docs
54ee9dd35363edb1c697e9e91792d5018ebaeda3 @losipiuk #13191 ✅ rn ✅ docs
0c316bdf294bb53dbad6e02b085a1890b2c8c248 @arhimondr #12963 ✅ rn ✅ docs
61efa644b20c824635a220cdec84f6a9552a7045 @arhimondr #12963 ✅ rn ✅ docs
56cb0f40c833f878e743ccb15a51a10914bb6db5 @arhimondr #12963 ✅ rn ✅ docs
47205f053cf210b6e2a36d5c4b27a02c467156c5 @arhimondr #12963 ✅ rn ✅ docs
4a2217f5aea3e1b381fc9065eb84a379726e801b @arhimondr #12963 ✅ rn ✅ docs
850db66e2d5bf7be774540a5860456369e9e1dcc @arhimondr #12963 ✅ rn ✅ docs
145ec3106b203099c6ce45ccd3f03931f7150211 @arhimondr #12963 ✅ rn ✅ docs
b3d38c3a06518d08323997fd37fbb89bc9979a76 @arhimondr #13055 ✅ rn ✅ docs

## 18 July 2022 ✅ 

c7d3c757efa9ce129709ec569595a455eddd699c @raunaqmorarka #13203 ✅ rn ✅ docs
75cf1a89526893cac1a30017505e73dee3a238bc @losipiuk #10767 ✅ rn ✅ docs
b21363abba702f13d39839fb1c5cc99bde02da38 @losipiuk #10767 ✅ rn ✅ docs
459474d4f54d9539c43d303139449e6d50da565c @dain #13067 ✅ rn ✅ docs
b22e599d08340c7c1de4595908f986bd831f034b @dain #13067 ✅ rn ✅ docs
9bc2f6c75d8f3d9d93f654349f1bbe41eb9266d9 @ebyhr #13164 ✅ rn ✅ docs

## 19 July 2022 ✅

69c47659f84d2501237a49f1d37e75a826664bac @kokosing #13127 ✅ rn ✅ docs
76c02ee4ec6cabe725cdd25ac870f9de32a31be7 @kokosing #13127 ✅ rn ✅ docs
e5bf35345a046ed8dd9fda74a42649b1fc2bd854 @ebyhr #13105 ✅ rn ✅ docs
0beb0e0ab93315976b4d5ef49fe555a3139cc12a @ebyhr #13105 ✅ rn ✅ docs
345652efced5f6c9b3491adff91156b67df167fb @sopel39 #12548 ✅ rn ✅ docs
ada381f9a1e877c6c52ffaeedf7598180aa1c6e8 @findepi #13207 ✅ rn ✅ docs
49003d5843600523cb1c58cf4f0d5bb9d0a9d1d3 @sopel39 #13213 ✅ rn ✅ docs
1abdc8f8a7516a48666cbb477f3b945834acf460 @raunaqmorarka #13208 ✅ rn ✅ docs
3a0491daad79585106f3e07d1195027b1c7d61e3 @raunaqmorarka #13208 ✅ rn ✅ docs
822f43e00c2db438291cbcbc4dac5c64cd426766 @raunaqmorarka #13208 ✅ rn ✅ docs
35f68312ed7dc065245d4d55d9420e10d74069f8 @raunaqmorarka #13208 ✅ rn ✅ docs
12127a368a01f486aab9879819e0fd2a00004bca @electrum #11677 ✅ rn ✅ docs
dd60fe184e4f44b2967b37256f07ae0c5a1fff9a @phd3 #12392 ✅ rn ✅ docs
72ac665ca73d97f36c252ee32b0b093bfc32ace9 @phd3 #12392 ✅ rn ✅ docs

## 20 July 2022 ✅ 

a2c8f986523f1e6ab9a0eb9363b90ac8f029f98b @electrum #12895 ✅ rn ✅ docs
a5f9e860d8f2182a18be1b2f12544c01f85b7a7b @ebyhr #13122 ✅ rn ✅ docs
effccfc818cb9fd6ed46bf54174eaa366344f745 @ebyhr #13122 ✅ rn ✅ docs
f01510d7eace82dd34439f53caf4ba5b42224c85 @ebyhr #13122 ✅ rn ✅ docs
305c486a31ab7cbce974d94e16aed958af34019c @ebyhr #13122 ✅ rn ✅ docs
767e6bb3938cad1bdd4a1a531b31d6f83939aa0c @ebyhr #13122 ✅ rn ✅ docs
55261cfed79bbc17402e91c0cac3faf2639bb82c @ebyhr #13240 ✅ rn ✅ docs
7275866f9cb7d0e940fe057f6366024362cb22d1 @findepi #13209 ✅ rn ✅ docs
1a044c36ade475b20fe52af88631c874ff993174 @raunaqmorarka #13226 ✅ rn ✅ docs
308e717baf78dc344d7f962bf32155416b751902 @raunaqmorarka #13233 ✅ rn ✅ docs
a7455f23d246585f71ead05dde1749380763cfa0 @findepi #12975 ✅ rn ✅ docs
44bb41a50a4000ea8751036f62c52113f07d5453 @raunaqmorarka #13125 ✅ rn ✅ docs
5205b40e91fb1980528379344d5a0ec030de5a28 @findepi #13078 ✅ rn ✅ docs
e246dfa0953c6615cb1338bb84de503c4109a3b0 @findepi #13078 ✅ rn ✅ docs
c16175d63e0ac21d012c588777059115408e2fa5 @findepi #13154 ✅ rn ✅ docs
b9487c951600e92fb7d997807a214be98ef5afcf @findepi #13229 ✅ rn ✅ docs
8112491cab3c4928224efa9dc738d6e1dfd27560 @findepi #13227 ✅ rn ✅ docs
4b199959dd390828902277b3e13c165271ed2df0 @findepi #13250 ✅ rn ✅ docs
6f34eedf55818a00c76e8791eadb8c735c81e040 @findepi #13253 ✅ rn ✅ docs
791872c212eb370dd385669bf082f9ce951beb42 @electrum #13223 ✅ rn ✅ docs
b9e0a40fd948a2aa0d5c8aeb6af2a908d92407db @electrum #13206 ✅ rn ✅ docs
2262123bdff0d601ff3f9b6bf4fc78febda79e0d @arhimondr #13217 ✅ rn ✅ docs
ea28b6a84da8cafa06346be2d2c05a27d7dcbf8b @arhimondr #13217 ✅ rn ✅ docs
d2ffc3bc848f583cac1aace1df9ad1e80493e393 @ebyhr #13252 ✅ rn ✅ docs
4aaf50599308e5818af3c86beee8222c5f5f19b2 @kokosing #12722 ✅ rn ✅ docs
1ae25e2df83df04239c9e2527e84e92c697a5a5f @electrum #13236 ✅ rn ✅ docs
4d15cc6009691fce7ddaf9d31dde9123b26d9bad @electrum #13236 ✅ rn ✅ docs
2b636073232a498b6f71ed88399aaf0a7e4bf584 @electrum #13236 ✅ rn ✅ docs
649bd3a358db35e612c71ab1dd579032afd3c022 @electrum #13236 ✅ rn ✅ docs
6ba50354879da343d9499044952581eb93c41d9c @electrum #13236 ✅ rn ✅ docs
bf3d164d43efb38ec2ad2a2a05e42d142df2fb57 @electrum #13236 ✅ rn ✅ docs

## 21 July 2022

8c319ac74ffa9a9c60e224e3f8a6887b91e6a978 @raunaqmorarka #13263 ✅ rn ✅ docs
646191561caae6e80e7ba70bfbaea564366516a7 @raunaqmorarka #13249 ✅ rn ✅ docs
f28c507ca2222b864324a2327eda3f874b1474e8 @sopel39 #13268 ✅ rn ✅ docs
94a87dac3cfc1feb85c6c42ec11f74ad83aa0fa3 @sopel39 #13268 ✅ rn ✅ docs
1383ea94c129000c306cc7454cd61faffcb3a2a4 @findepi #13034 ✅ rn ✅ docs
0dc71139a5f23660cf14b058c7eb6767bbcfa910 @findepi #13034 ✅ rn ✅ docs
01843af522e64ed6fc6d5415eb261923fd28a28b @findepi #13034 ✅ rn ✅ docs
03a23e4b0aa8ab1d4143416e6c777a765b0b2d5d @findepi #13034 ✅ rn ✅ docs
6a4fcc04ccfddefc5f033c0aeec76bd01366821f @ebyhr #13195 ✅ rn ✅ docs
a2bbb2e5ab7860617f83ca1dd1135bfe8aa9eb3c @findepi #13047 ✅ rn ✅ docs
53bd06443d4550f8576e21bf918ac7ded946e685 @findepi #13047 ✅ rn ✅ docs
4808b16faaabc6c4cd2139a6115c9f3145a89b3e @ebyhr #13242 ✅ rn ✅ docs
91e570ed4952b91d36c6ed4417d4b4ce2dfd0939 @ebyhr #13242 ✅ rn ✅ docs
d1feda897e0064cb229fe9552a0d91cd5bd1651e @martint #13284 ✅ rn ✅ docs
855ee039193d3bf7c704e7c582d9fdf209ed10b2 @electrum #13222 ✅ rn ✅ docs
fe4b939170407520fb3bb487e527cd9fe177786a @electrum #13155 ✅ rn ✅ docs
9b24cf034f693f5af288b3cc2e31c6cc4c333b86 @electrum #13155 ✅ rn ✅ docs
55187ffc55bd20c3ff4ec19fd092ac3024b9a3b4 @electrum #13155 ✅ rn ✅ docs
68240751c9fa933d2b700f9f508dce63a4b2e691 @sopel39 #13292 ✅ rn ✅ docs
c9a5006efbb4d1a278d5a4552e97a0946fc42f2b @sopel39 #13290 ✅ rn ✅ docs
d0c6cd1c3ae36c5c8576166e6d23d0f6065240fa @sopel39 #13290 ✅ rn ✅ docs
cdcbe3642944c8af5485915494922a8821c1abfa @martint #13224 ✅ rn ✅ docs
d8b3e5d525cb8a35a6814708617414465d87b247 @martint #13238 ✅ rn ✅ docs
da4fcea0ab6a0b6ab6a9eef339c49a35637fa8cf @kasiafi #13228 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-391.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+## BigQuery connector
+
+* Add support for reading external tables. ({issue}`13164`)
+* Add support for specifying a table comment and column comments when creating a
+  table. ({issue}`13105`)
+
+## Hive connector
+
+* Add support for
+  [AWS Athena partition projection](https://docs.aws.amazon.com/athena/latest/ug/partition-projection.html). ({issue}`11305`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-391.md:16]: Lets not link there... instead we have to make sure the documentation PR is merged and then we link there.

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-391.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Release 391 (20 July 2022)
+
+## General
+
+* Improve performance of repeated aggregations with ``CASE`` expressions. ({issue}`12548`)
+* Improve query performance when calling ``setBytes`` within a
```
Comment (martint) [inline docs/src/main/sphinx/release/release-391.md:6]: "synchronization section" is not something users understand. What are the user-visible conditions under which performance will improve? @sopel39

File: docs/src/main/sphinx/release/release-391.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  ``synchronization`` section. ({issue}`13213`)
+* Fail queries that get stuck in long-running regular expression functions. ({issue}`12392`)
+* Set the default value of the ``optimizer.force-single-node-output``
+  configuration property to false. ({issue}`13217`)
+
+## BigQuery connector
+
+* Add support for reading external tables. ({issue}`13164`)
+* Add support for specifying a table comment and column comments when creating a
```
Comment (martint) [inline docs/src/main/sphinx/release/release-391.md:15]: > ... specifying table and column comments ...

File: docs/src/main/sphinx/release/release-391.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+# Delta Lake connector
+
+* Fix incorrect results when using the Glue metastore with queries that contain
+  `IS NULL` and additional filters. ({issue}`13122`)
+
+## Hive connector
+
+* Add support for AWS Athena partition projection. ({issue}`11305`)
+* Fix creating empty ORC bucket files when using ZSTD. ({issue}`9775`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-391.md:26]: What's the fix? What was broken? How does it impact users (was it a failure, corrupted table, etc)?

File: docs/src/main/sphinx/release/release-391.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+# Iceberg connector
+
+* Improve Parquet writer performance. ({issue}`13203`, {issue}`13208`)
+* Improve query planning performance. ({issue}`13047`)
+* Fix query failure when reading an Iceberg table with deletion-tracking files. ({issue}`13035`)
+
+# MariaDB connector
+
+* Fix potential query failure when using the ``query`` table function with
+  metadata caching and the underlying table's schema is changed via Trino. ({issue}`12526`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-391.md:52]: > ... underlying table schema ...

File: docs/src/main/sphinx/release/release-391.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+* Fix creating empty ORC bucket files when using ZSTD. ({issue}`9775`)
+* Fix query failure or potentially incorrect statistics when running concurrent
+  `CREATE TABLE AS` queries with the `IF NOT EXISTS` clause for the same
+  non-existent table. ({issue}`12895`)
+* Fix incorrect results when using the Glue metastore with queries that contain
+  `IS NULL` and additional filters. ({issue}`13122`)
+
+# Iceberg connector
+
+* Improve Parquet writer performance. ({issue}`13203`, {issue}`13208`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-391.md:45]: > Improve performance when writing Parquet data

File: docs/src/main/sphinx/release/release-391.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+* Improve query planning performance. ({issue}`13047`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-391.md:46]: This is an engine-wide improvement, so it should go in the general section. It should also say something like this:

> Improve planning performance for join queries when when tables have statistics.

File: docs/src/main/sphinx/release/release-391.md [hunk 8]
```diff
@@ -13,8 +13,7 @@
 ## BigQuery connector
 
 * Add support for reading external tables. ({issue}`13164`)
-* Add support for specifying a table comment and column comments when creating a
-  table. ({issue}`13105`)
+* Add support for specifying a table and column comments when creating a table. ({issue}`13105`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-391.md:16]: Remove "a" from "a table and column comments".

File: docs/src/main/sphinx/release/release-391.md [hunk 9]
```diff
@@ -34,7 +33,7 @@
 
 * Add support for AWS Athena partition projection. ({issue}`11305`)
 * Improve optimized Parquet writer performance. ({issue}`13203`, {issue}`13208`)
-* Fix creating empty ORC bucket files when using ZSTD. ({issue}`9775`)
+* Fix potential failure when creating empty ORC bucket files while using ZSTD. ({issue}`9775`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-391.md:36]: > ... while using ZSTD compression

**Pull request merged**